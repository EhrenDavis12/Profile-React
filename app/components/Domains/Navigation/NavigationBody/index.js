/**
 *
 * Navigation
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';
// import { Link } from 'react-router-dom';
import AppBar from 'components/Domains/Navigation/AppBar';
import Drawer from 'components/Domains/SharedKernel/Drawer';
import './index.css';

function NavigationBody({
  navigationLinks,
  toggleDrawer,
  selectItem,
  isDrawerOpen,
}) {
  /* const nav = (
    <nav>
      <ul>
        <li>
          <Link to="/">About Me</Link>
        </li>
      </ul>
    </nav>
  ); */
  return (
    <>
      <AppBar toggleDrawer={toggleDrawer} />
      <Drawer
        items={navigationLinks}
        selectItem={selectItem}
        itemLabelAttr="DisplayName"
        itemKeyAttr="key"
        isDrawerOpen={isDrawerOpen}
      />
    </>
  );
}

NavigationBody.propTypes = {
  // links: PropTypes.array.isRequired,
  navigationLinks: PropTypes.array.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
  selectItem: PropTypes.func.isRequired,
  isDrawerOpen: PropTypes.bool.isRequired,
};

export default NavigationBody;
