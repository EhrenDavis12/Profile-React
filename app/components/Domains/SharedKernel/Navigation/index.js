/**
 *
 * Navigation
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';
import { Link } from 'react-router-dom';
import './index.css';

function Navigation() {
  const nav = (
    <nav>
      <ul>
        <li>
          <Link to="/">About Me</Link>
        </li>
        <li>
          <Link to="/portfolio">Portfolio</Link>
        </li>
        <li>
          <Link to="/contact">Contact</Link>
        </li>
        <li>
          <Link to="/messages">Messages</Link>
        </li>
      </ul>
    </nav>
  );
  return nav;
}

Navigation.propTypes = {};

export default Navigation;
