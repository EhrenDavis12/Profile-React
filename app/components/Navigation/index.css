ul.unstyled {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

nav ul {
  opacity: 0.8;
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-image: linear-gradient(
    to bottom right,
    rgb(8, 253, 253),
    rgb(0, 26, 255)
  );
}

nav li {
  float: left;
}

nav li a {
  display: block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

nav li a:hover {
  background-color: rgb(8, 253, 253);
  color: white;
}

nav button {
  margin-top: 13px;
}
