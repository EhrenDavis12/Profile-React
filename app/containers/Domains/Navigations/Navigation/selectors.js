import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the Navigation state domain
 */

const selectNavigationDomain = state => state.get('Navigation', initialState);

/**
 * Other specific selectors
 */

/**
 * Default selector used by Navigation
 */

const makeSelectNavigation = () =>
  createSelector(selectNavigationDomain, substate => substate.toJS());

export default makeSelectNavigation;
export { selectNavigationDomain };
