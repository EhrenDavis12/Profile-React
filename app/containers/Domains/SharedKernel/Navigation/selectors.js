import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the navigation state domain
 */

const selectNavigationDomain = state => state.get('navigation', initialState);

/**
 * Other specific selectors
 */

/**
 * Default selector used by Navigation
 */

const makeSelectNavigation = () =>
  createSelector(selectNavigationDomain, substate => substate.toJS());

export default makeSelectNavigation;
export { selectNavigationDomain };
