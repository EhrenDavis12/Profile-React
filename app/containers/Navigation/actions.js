/*
 *
 * Navigation actions
 *
 */

import { TOGGLE_DRAWER, SELECT_LINK } from './constants';

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER,
  };
}

export function selectLink(selectedLink) {
  return {
    type: SELECT_LINK,
    selectedLink,
  };
}
