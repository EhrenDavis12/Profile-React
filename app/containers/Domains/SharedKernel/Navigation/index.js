/**
 *
 * Navigation
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

// import NavigationComponents from 'components/Domains/SharedKernel/Navigation';
import AppBar from 'components/Domains/SharedKernel/AppBar';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectNavigation from './selectors';
import reducer from './reducer';
import saga from './saga';
import { toggleDrawer } from './actions';

/* eslint-disable react/prefer-stateless-function */
export class Navigation extends React.Component {
  render() {
    return (
      <div>
        <AppBar {...this.props} />
      </div>
    );
  }
}

Navigation.propTypes = {
  // dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  navigation: makeSelectNavigation(),
});
// const mapStateToProps = createStructuredSelector();

function mapDispatchToProps(dispatch) {
  return {
    toggleDrawer: () => dispatch(toggleDrawer()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'navigation', reducer });
const withSaga = injectSaga({ key: 'navigation', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Navigation);
