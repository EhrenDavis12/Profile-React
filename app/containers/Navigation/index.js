/**
 *
 * Navigation
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import NavigationBody from 'components/Navigation/NavigationBody';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectNavigation from './selectors';
import reducer from './reducer';
import saga from './saga';
import { toggleDrawer, selectLink } from './actions';

let lastScrollY = 0;
let ticking = false;

/* eslint-disable react/prefer-stateless-function */
export class Navigation extends React.Component {
  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll, true);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  nav = React.createRef();

  handleScroll = () => {
    lastScrollY = window.scrollY;

    if (!ticking) {
      window.requestAnimationFrame(() => {
        this.nav.current.style.top = `${lastScrollY}px`;
        ticking = false;
      });

      console.log('scroll Fire');
      ticking = true;
    }
  };

  render() {
    return (
      <div>
        <NavigationBody {...this.props} />
      </div>
    );
  }
}

Navigation.propTypes = {
  // dispatch: PropTypes.func.isRequired,
};

/* const mapStateToProps = createStructuredSelector({
  Navigation: makeSelectNavigation(),
}); */
const mapStateToProps = makeSelectNavigation();

export function mapDispatchToProps(dispatch) {
  return {
    toggleDrawer: () => dispatch(toggleDrawer()),
    selectItem: linkSelected => dispatch(selectLink(linkSelected)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'Navigation', reducer });
const withSaga = injectSaga({ key: 'Navigation', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Navigation);
