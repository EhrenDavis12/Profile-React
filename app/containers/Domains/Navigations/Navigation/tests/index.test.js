import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
// import { enzymeFind } from 'styled-components/test-utils';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../../../configureStore';

import Navigation from '../index';

describe('<Navigation />', () => {
  let store;
  let wrapper;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
    wrapper = renderComponentMount();
  });

  const renderComponentMount = () =>
    mount(
      <Provider store={store}>
        <Navigation />
      </Provider>,
    );
  // mount(<Navigation {...navigationLinksProps} />);

  describe('<Navigation /> snap Shots', () => {
    it('renders as expected with populated messages', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Navigation />', () => {
    it('should render its NavigationBody', () => {
      expect(wrapper.find('NavigationBody').length).toEqual(1);
    });
  });
});
