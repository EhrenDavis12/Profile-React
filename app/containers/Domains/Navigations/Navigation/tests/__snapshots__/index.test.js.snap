// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> <Navigation /> snap Shots renders as expected with populated messages 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "Navigation": [Function],
      },
      "injectedSagas": Object {
        "Navigation": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": Array [],
            "name": "rootSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(withSaga(Connect(Navigation)))>
    <withSaga(Connect(Navigation))>
      <Connect(Navigation)>
        <Navigation
          isDrawerOpen={false}
          navigationLinks={
            Array [
              Object {
                "DisplayName": "About Me",
                "LinkTo": "/",
                "key": "AboutMe",
              },
              Object {
                "DisplayName": "Portfolio",
                "LinkTo": "/portfolio",
                "key": "Portfolio",
              },
              Object {
                "DisplayName": "Contact",
                "LinkTo": "/contact",
                "key": "Contact",
              },
              Object {
                "DisplayName": "Messages",
                "LinkTo": "/messages",
                "key": "Messages",
              },
            ]
          }
          selectItem={[Function]}
          toggleDrawer={[Function]}
        >
          <div>
            <NavigationBody
              isDrawerOpen={false}
              navigationLinks={
                Array [
                  Object {
                    "DisplayName": "About Me",
                    "LinkTo": "/",
                    "key": "AboutMe",
                  },
                  Object {
                    "DisplayName": "Portfolio",
                    "LinkTo": "/portfolio",
                    "key": "Portfolio",
                  },
                  Object {
                    "DisplayName": "Contact",
                    "LinkTo": "/contact",
                    "key": "Contact",
                  },
                  Object {
                    "DisplayName": "Messages",
                    "LinkTo": "/messages",
                    "key": "Messages",
                  },
                ]
              }
              selectItem={[Function]}
              toggleDrawer={[Function]}
            >
              <AppBar
                toggleDrawer={[Function]}
              >
                <div
                  className="appBar"
                >
                  <BasicButton
                    className="iconButton"
                    onClick={[Function]}
                  >
                    <div
                      className="iconButton"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <FontAwesome
                        className="icon"
                        key=".0"
                        name="bars"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-bars icon"
                        />
                      </FontAwesome>
                    </div>
                  </BasicButton>
                  <div
                    className="heading"
                  >
                    Coder Daily
                  </div>
                  <div
                    className="linkContainer"
                  >
                    Log In
                  </div>
                </div>
              </AppBar>
              <Drawer
                isDrawerOpen={false}
                itemKeyAttr="key"
                itemLabelAttr="DisplayName"
                items={
                  Array [
                    Object {
                      "DisplayName": "About Me",
                      "LinkTo": "/",
                      "key": "AboutMe",
                    },
                    Object {
                      "DisplayName": "Portfolio",
                      "LinkTo": "/portfolio",
                      "key": "Portfolio",
                    },
                    Object {
                      "DisplayName": "Contact",
                      "LinkTo": "/contact",
                      "key": "Contact",
                    },
                    Object {
                      "DisplayName": "Messages",
                      "LinkTo": "/messages",
                      "key": "Messages",
                    },
                  ]
                }
                selectItem={[Function]}
              >
                <div
                  className="drawer"
                >
                  <BasicButton
                    className="item"
                    key="AboutMe"
                    onClick={[Function]}
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      About Me
                    </div>
                  </BasicButton>
                  <BasicButton
                    className="item"
                    key="Portfolio"
                    onClick={[Function]}
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      Portfolio
                    </div>
                  </BasicButton>
                  <BasicButton
                    className="item"
                    key="Contact"
                    onClick={[Function]}
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      Contact
                    </div>
                  </BasicButton>
                  <BasicButton
                    className="item"
                    key="Messages"
                    onClick={[Function]}
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      Messages
                    </div>
                  </BasicButton>
                </div>
              </Drawer>
            </NavigationBody>
          </div>
        </Navigation>
      </Connect(Navigation)>
    </withSaga(Connect(Navigation))>
  </withReducer(withSaga(Connect(Navigation)))>
</Provider>
`;
